modules/market/src/main/java/com/opengamma/strata/market/curve/definition/FxSwapCurveNode.java

```diff
@@ -89,9 +92,10 @@ public DatedCurveParameterMetadata metadata(LocalDate valuationDate) {
 
   @Override
   public FxSwapTrade trade(LocalDate valuationDate, MarketData marketData) {
-    double fxNearRate = marketData.getValue(fxNearKey);
+    FxRate fxNearRate = marketData.getValue(fxNearKey);
+    double rate = fxNearRate.fxRate(fxNearKey.getPair());
```

Comment:
This needs to use the convention's currency pair to extract the data to ensure that the rate is the right way round.